store/tikv/txn.go

```diff
@@ -245,6 +245,7 @@ func (txn *tikvTxn) LockKeys(keys ...kv.Key) error {
 	for _, key := range keys {
 		txn.lockKeys = append(txn.lockKeys, key)
 	}
+	txn.dirty = true
```

Comment:
Is this a bug fix?

Reply:
If we add the optimization that return early if read only, then this is needed.

Reply:
got it